src/Logger/BaseTextFileLogger.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Logger;
+
+use function array_map;
+use function explode;
+use function implode;
+use Infection\Framework\Str;
+use Infection\Metrics\ResultsCollector;
+use Infection\Mutant\MutantExecutionResult;
+use const PHP_EOL;
+use function sprintf;
+
+/**
+ * @internal
+ */
+abstract readonly class BaseTextFileLogger implements LineMutationTestingResultsLogger
+{
+    public function __construct(
+        private ResultsCollector $resultsCollector,
+        private bool $debugVerbosity,
+        private bool $onlyCoveredMode,
+        private bool $debugMode,
+    ) {
+    }
+
+    public function getLogLines(): array
+    {
+        $separateSections = false;
+
+        $logs = [];
+
+        $notes = [];
+
+        if (!$this->debugVerbosity) {
+            $notes[] = 'Note: Pass `--log-verbosity=all` to log information about killed and errored mutants.';
+        }
+
+        if (!$this->debugMode) {
+            $notes[] = 'Note: Pass `--debug` to log test-framework output.';
+        }
+
+        if ($notes !== []) {
+            foreach ($notes as $note) {
```

Comment:
is there a reason why we have `$notes` besides `$logs` here? They are both initialised at the beginning so we could instead just have `$logs`.

Reply:
Let's not bother here it's fine really. It was more an illustration to a point @maks-rafalko made the other day

Reply:
I absolutely understand when the split is needed, but there was a case were it was not needed :)